### ISSUE TITLE
Updates to ICOB API's

### ISSUE THREAD

**Issue by ianco** [bcgov/indy-catalyst#344]
Make the following updates to the ICOB API's:

1.  Add support for API version via Accept headers
     - work is in progress, also maintains version support within the URL

2.  Remove internal ID's from the URL
     - only include attributes that are publicly known and don't change, such as the source_id (corp_num), cardinality fields for credentials, etc

3.  In API response data, hide details of the underlying ICOB data model and make the response data more "credential-like"

4.  Add support for issuer-defined data (as described in some other ticket)

**Comment by swcurran** [comment]
Status update @esune @ianco ?  Move to backlog?  I think there are some related issues to this?

**Comment by esune** [comment]
#260 has an open PR with changes for the API header/path versioning. Once that is merged we will be able to create a new v3 API with the restructured data structures.

@swcurran I moved back #260 to review/QA since the PR has not yet been merged.

**Comment by swcurran** [comment]
Sorry about moving it. I sometimes miss when something is a PR vs. an issue.

**Comment by esune** [comment]
The API versioning part is done. Not sure on whether we want to keep this ticket open for the data restructuring, or we should log a new ticket and tackle it when the time comes.

@ianco @nrempel I believe Nick will be working on the repo refactoring and we agreed that the data refactoring step may come just after that,am I correct?
